src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -381,3 +383,85 @@ bool VtolType::is_channel_set(const int channel, const int target)
 
 	return (channel_bitmap >> channel) & 1;
 }
+
+
+void VtolType::pusher_assist()
+{
+
+	// normalized pusher throttle (standard VTOL) or tilt (tiltrotor), initialize to 0
+	_hover_pusher_or_tilt_forward_actuation = 0.0f;
+
+	// if the thrust scale param is zero or the drone is on manual mode,
+	// then the pusher-for-pitch strategy is disabled and we can return
+	if (_params->forward_thrust_scale < FLT_EPSILON ||
+	    !_v_control_mode->flag_control_position_enabled) {
+		return;
+	}
+
+	// Do not engage pusher assist during a failsafe event
+	// There could be a problem with the fixed wing drive
+	if (_attc->get_vtol_vehicle_status()->vtol_transition_failsafe) {
+		return;
+	}
+
+	// disable pusher assist during landing
+	if (_attc->get_pos_sp_triplet()->current.valid
+	    && _attc->get_pos_sp_triplet()->current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
+		return;
+	}
+
+	const Dcmf R(Quatf(_v_att->q));
+	const Dcmf R_sp(Quatf(_v_att_sp->q_d));
+	const Eulerf euler(R);
+	const Eulerf euler_sp(R_sp);
+
+	// direction of desired body z axis represented in earth frame
+	Vector3f body_z_sp(R_sp(0, 2), R_sp(1, 2), R_sp(2, 2));
+
+	// rotate desired body z axis into new frame which is rotated in z by the current
+	// heading of the vehicle. we refer to this as the heading frame.
+	Dcmf R_yaw = Eulerf(0.0f, 0.0f, -euler(2));
+	body_z_sp = R_yaw * body_z_sp;
+	body_z_sp.normalize();
+
+	// calculate the desired pitch seen in the heading frame
+	// this value corresponds to the amount the vehicle would try to pitch forward
```

Comment:
@sfuhrer Let's call it "pitch down" instead of "pitch forward". Not something that you wrote though.